x-pack/legacy/plugins/ml/public/data_frame/components/define_pivot/define_pivot_form.tsx

```diff
@@ -352,62 +465,192 @@ export const DefinePivotForm: SFC<Props> = React.memo(({ overrides = {}, onChang
+                    options={aggOptions}
+                    placeholder={i18n.translate(
+                      'xpack.ml.dataframe.definePivotForm.aggregationsPlaceholder',
+                      {
+                        defaultMessage: 'Add an aggregation ...',
+                      }
+                    )}
+                  />
+                </Fragment>
+              </EuiFormRow>
             </Fragment>
-          </EuiFormRow>
+          )}
 
-          <EuiFormRow
-            label={i18n.translate('xpack.ml.dataframe.definePivotForm.aggregationsLabel', {
-              defaultMessage: 'Aggregations',
-            })}
-          >
+          {isAdvancedEditorEnabled && (
             <Fragment>
-              <AggListForm
-                list={aggList}
-                options={aggOptionsData}
-                onChange={updateAggregation}
-                deleteHandler={deleteAggregation}
-              />
-              <DropDown
-                changeHandler={addAggregation}
-                options={aggOptions}
-                placeholder={i18n.translate(
-                  'xpack.ml.dataframe.definePivotForm.aggregationsPlaceholder',
-                  {
-                    defaultMessage: 'Add an aggregation ...',
-                  }
-                )}
-              />
+              <EuiFormRow
+                label={i18n.translate('xpack.ml.dataframe.definePivotForm.advancedEditorLabel', {
+                  defaultMessage: 'Pivot configuration object',
+                })}
+                helpText={advancedEditorHelpText}
+              >
+                <EuiPanel grow={false} paddingSize="none">
+                  <EuiCodeEditor
+                    mode="json"
+                    width="100%"
+                    value={advancedEditorConfig}
+                    onChange={(d: string) => {
+                      setAdvancedEditorConfig(d);
+
+                      // Disable the "Apply"-Button if the config hasn't changed.
+                      if (advancedEditorConfigLastApplied === d) {
+                        setAdvancedEditorApplyButtonEnabled(false);
+                        return;
+                      }
+
+                      // Try to parse the string passed on from the editor.
+                      // If parsing fails, the "Apply"-Button will be disabled
+                      try {
+                        JSON.parse(d);
+                        setAdvancedEditorApplyButtonEnabled(true);
+                      } catch (e) {
+                        setAdvancedEditorApplyButtonEnabled(false);
+                      }
+                    }}
+                    setOptions={{
+                      fontSize: '12px',
+                    }}
+                    aria-label={i18n.translate(
+                      'xpack.ml.dataframe.definePivotForm.advancedEditorAriaLabel',
+                      {
+                        defaultMessage: 'Advanced Editor',
```

Comment:
Should be sentence case here `Advanced editor`

Reply:
Fixed in 55cb857.